lib/ansible/modules/crypto/luks_device.py

```diff
@@ -474,12 +467,15 @@ def run_module():
 
     # luks create
     if conditions.luks_create():
-        try:
-            crypt.run_luks_create(module.params['device'],
-                                  module.params['keyfile'])
-        except ValueError as e:
-            module.fail_json(msg="luks_device error: %s" % e)
+        if not module.check_mode:
+            try:
+                crypt.run_luks_create(module.params['device'],
+                                      module.params['keyfile'])
+            except ValueError as e:
+                module.fail_json(msg="luks_device error: %s" % e)
         result['changed'] = True
```

Comment:
What happens if the device already exists?
Seems we get success but when we'll run it not in check mode it will fail.
Maybe we'd better add some checks, hadn't we?
For example, by the following, etc
```
    def is_luks(self, device):
        ''' check if the LUKS device does exist
        '''
        result = self._run_command([self._cryptsetup_bin, 'isLuks', device])
        return result[RETURN_CODE] == 0
```
If the device doesn't exist we can set result['changed'] = True, else (if exists) = False

Reply:
That the device exists (if it is specified / required for the operation) is a requirement for the module. I guess it should be checked before anything else is done (and the module should fail if it doesn't).

Reply:
Hmm, "LUKS device does exist" is a strange way to say "device is a LUKS device". When I say "device exists", I mean that `device` is actually a device. That function docstring should really be reformulated.

Reply:
I've fixed the docstring and added a check whether `device` actually points to a device.